### ISSUE TITLE
Feature Request: Automatically reshim after a mix escript.install command successfully finishes

### ISSUE THREAD

**Issue by axelson** [asdf-vm/asdf-elixir#103]
The current Livebook instructions say that if you're using asdf you need to run `asdf reshim elixir` after installing livebook as an escript:
> Running Livebook using Escript makes for a very convenient option
> for local usage and provides easy configuration via CLI options.
>
> ```shell
> mix escript.install hex livebook
>
> # Start the Livebook server
> livebook server
>
> # See all the configuration options
> livebook server --help
> ```
>
> After you install the escript, make sure you add the directory where
> Elixir keeps escripts to your [$PATH](https://en.wikipedia.org/wiki/PATH_(variable)).
> If you installed Elixir with `asdf`, you'll need to run `asdf reshim elixir`
> once the escript is built.
Source:
https://github.com/elixir-nx/livebook/blob/26226a42e0544fe3870c29bc68d752e7d6d3e63c/README.md#escript

Is it possible for asdf-elixir to detect that `asdf escript.install` has successfully installed an escript and then automatically reshim? If not that's fine, just figured I'd ask to remove a potential stumbling block. Escripts are always executable binaries so reshiming makes sense (to me). We'd probably have to do it when uninstalling an escript as well.

**Comment by Stratus3D** [comment]
@axelson @kamidev @hubertlepicki, I don't have time to work on this myself right now, but a PR would be welcome. I think the asdf-nodejs plugin does something along these lines with the custom npm shim - https://github.com/asdf-vm/asdf-nodejs/blob/master/shims/npm That npm shim may not be working at the moment, but it should be possible to do something similar and get it working for `mix deps.get`.

**Comment by kamidev** [comment]
I'll take a  look at this. Perhaps a separate shim for escript would be enough? Or is there a reason we need to check mix.deps.get, too?

**Comment by axelson** [comment]
I don't think we need to reshim after a `mix deps.get` because that won't add any global binaries. I think we'd only need to automatically reshim after `mix escript.install` and `mix escript.uninstall`.

**Comment by kamidev** [comment]
That's what I was thinking, too. I am playing around trying to get a forked project with an 'escript' shim (based on the npm example above) to run. I'll keep you updated if I make any progress.

**Comment by axelson** [comment]
Sounds good!
